### ISSUE TITLE
Empty equation with spaces is not recognized as block equation

### ISSUE THREAD

**Issue by Enter-tainer** [typst/typst#4791]
### Description

```typ
$  $
```
The ast for code above is
```
Markup: 5 [
    Equation: 4 [
        Dollar: "$",
        Math: 0,
        Space: "  ",
        Dollar: "$",
    ],
    Space: "\n",
]
```
And for some reason it is not considered as block equation. I guess it is related some edge case in parser so it is not correctly recognized.
```
[src\pretty\mod.rs:331:9] equation.block() = false
```

### Operating system

Web app, Windows, Linux, macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enter-tainer** [comment]
https://github.com/Myriad-Dreamin/tinymist/issues/548

**Comment by wrzian** [comment]
This is handled by this in the `ast.rs` file:

https://github.com/typst/typst/blob/a30b681251ef684b4bcfdc43787ad058b5e600f6/crates/typst-syntax/src/ast.rs#L689-L695

**Comment by wrzian** [comment]
Seems that the way trivia is handled the spaces get combined and the empty `SyntaxKind::Math` gets in the way. If the empty math kind were removed the function would work as expected. The `body` function in the AST already returns a default none content, so only the equation parser would need to be updated.

**Comment by Enter-tainer** [comment]
yes. I guess we can add some special handling for this case.. For `$  $` or `$\n$` I think it is definitely block equation. But I cannot really tell if `$ $` is block or inline.
